src/components/FlightDetails/FlightItineraryInfo.tsx

```diff
@@ -0,0 +1,101 @@
+                    </Box>
+                </Grid>
+
+                {/* card info */}
+                <Grid item container direction={"row"} marginTop={1}>
+                    <Grid xs={2} item container marginTop={1} alignItems={"start"} justifyContent={"flex-end"}>
+                        <img alt="airlineIcon" src={airlineFailoverLogo} width={35} style={{marginRight: "1em"}} />
+                    </Grid>
+                    <Grid item xs={10}>
+                        <Grid item container xs={12} direction={"column"}>
+                            <Grid item xs={1}>
+                                <Typography variant={"body2"} className={flightCardStyles.x}>
+                                    {airlines?.get(segment.carrierCode)}
+                                </Typography>
+                            </Grid>
+                            <Grid item xs={1}>
+                                <Typography variant={"body2"} gutterBottom>
+                                    {timeTravelDiff(segment.departure.at, segment.arrival.at)}
+                                </Typography>
+                            </Grid>
+                            <Grid item xs={1}>
+                                <Box>
+                                    <LinearProgress variant="determinate" value={90} className={flightCardStyles.flightProgress}/>
+                                </Box>
+                            </Grid>
+                            <Grid item xs={1}>
+                                <Typography variant={"body2"} gutterBottom>
+                                    {moment(segment.departure.at).format('MMM DD,ddd')}
+                                </Typography>
+                            </Grid>
+                            <Grid item xs={1}>
+                                <Typography variant={"body2"} gutterBottom>
+                                    {convertDate(segment.departure.at)} - {convertDate(segment.arrival.at)}
+                                </Typography>
+                            </Grid>
+                            <Grid item xs={1}>
+                                <Typography variant={"body2"} gutterBottom>
+                                    {segment.departure.iataCode} → {segment.arrival.iataCode}
+                                </Typography>
+                            </Grid>
+                            <Grid item xs={1}>
+                                <Typography variant={"body2"} gutterBottom>
+                                    Passengers: {passengers}
+                                </Typography>
+                            </Grid>
+                        </Grid>
+                    </Grid>
+                </Grid>
+            </Grid>
+        );
+    });
+
+    return (
+        <Grid container data-testid={"flightItInfoComp"}>
+            {Elements}
+        </Grid>
+    );
+}
+
+export const buildMapOfCodes = (data: IAirlineInfo[]) => {
```

Comment:
Lets move this function to its own utils file so we leave the component clean and easy to read.

Reply:
Done✅